### PR TITLE
Replace basestring with future.utils.string_types.

### PR DESCRIPTION
**Pull request by rodrigc**
basestring is gone on Python 3.
On Python 2, future.utils.string_types expands to basestring.
On Python 3, future.utils.string_types expands to str.

### CODE REVIEW

**Comment by rutsky** [comment]
LGTM.

**Comment by rutsky** [comment]
@rodrigc when you will finish Python 3 porting please don't forget to add release notes for your changes to https://github.com/buildbot/buildbot/blob/master/master/docs/relnotes/index.rst

File: master/buildbot/process/buildstep.py [hunk 1]
```diff
@@ -1229,7 +1231,7 @@ def regex_log_evaluator(cmd, _, regexes):
         # we won't be changing "worst" unless possible_status is worse than it,
         # so we don't even need to check the log if that's the case
         if worst_status(worst, possible_status) == possible_status:
-            if isinstance(err, (basestring)):
+            if isinstance(err, (string_types)):
```
Comment (Stibbons) [inline master/buildbot/process/buildstep.py]: why there is parenthesis ? This does not add anything

File: master/buildbot/config.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 from future.utils import iteritems
 from future.utils import itervalues
+from future.utils import string_types
```
Comment (Stibbons) [inline master/buildbot/config.py]: This [do](http://python-future.org/compatible_idioms.html#basestring)c recommends to use:
```
from past.builtins import basestring
```

which one is better?

File: master/buildbot/process/buildstep.py (hunk 1 above)
Comment (rodrigc) [inline master/buildbot/process/buildstep.py]: You are correct, the parenthesis is not needed here.  It was there before my change,
but it wasn't needed before.  I can take it out.  That is not a problem.

File: master/buildbot/config.py (hunk 2 above)
Comment (rodrigc) [inline master/buildbot/config.py]: future.utils.string_types does the job quite nicely here, and I am confident that its usage
here is correct.  The past module is to be avoided and used sparingly.  The past module is meant for people who have huge codebases and don't want to make lots of changes.

The strategy that I am using is to bring the buildbot codebase forward while still preserving behavior and not breaking things on Python 2.6/2.7.

Comment (Stibbons) [inline master/buildbot/config.py]: make sense, thx

**Comment by tardyp** [comment]
needs rebase

**Comment by rodrigc** [comment]
I rebased

**Comment by sa2ajj** [comment]
@rodrigc it seems that this PR needs another rebase.

**Pull request merged**